src/streamlink_cli/main.py

```diff
@@ -866,10 +866,11 @@ def setup_plugin_args(session, parser):
     plugin_args = parser.add_argument_group("Plugin options")
     for pname, plugin in session.plugins.items():
         defaults = {}
+        group = plugin_args.add_argument_group(pname.capitalize())
 
         for parg in plugin.arguments:
             if not parg.is_global:
-                plugin_args.add_argument(parg.argument_name(pname), **parg.options)
+                group.add_argument(parg.argument_name(pname), **parg.options)
```

Comment:
I've just noticed that Python's argparse doesn't properly format help texts when adding nested argument groups, as it only includes the first level of groups. This currently ignores all plugin arguments in the `--help` output.

That means that `argparse.ArgumentParser.format_help` needs to be overridden, which is a bit annoying:
https://github.com/python/cpython/blame/761c5a1ce477b82cca35a79917c88d72d57ed776/Lib/argparse.py#L2521-L2526

The reason why I've nested the plugin arguments this is that it's easier to navigate/browse on the rendered HTML docs. The rendered manpage of course also includes the nested argument groups.